builder/struct.go

```diff
@@ -26,8 +26,37 @@ func (*Struct) Build(gen Generator, ctx *MethodContext, sourceID *xtype.JenID, s
 
 	name := ctx.Name(target.ID())
 	ctx.SetErrorTargetVar(jen.Id(name))
-	stmt := []jen.Code{
-		jen.Var().Id(name).Add(target.TypeAsJen()),
+
+	stmt := []jen.Code{}
+
+	initStmt := jen.Var().Id(name).Add(target.TypeAsJen())
+
+	constructor := ctx.Conf.Constructor
+	if constructor != nil {
+		constructorTargetType := constructor.Target.T.String()
+		if constructor.Target.PointerType != nil && target.PointerType == nil {
+			// if target is not pointer and constructor result it is, match target against constructor inner type
+			constructorTargetType = constructor.Target.PointerInner.T.String()
+		}
+
+		if constructorTargetType != target.T.String() {
+			cause := fmt.Sprintf("Constructor method return type mismatches with target: %s != %s", constructorTargetType, target.T.String())
+			return nil, nil, NewError(cause)
+		}
+
+		callStmt, callReturnID, err := gen.CallMethod(ctx, constructor, nil, nil, target, nil)
+		if err != nil {
+			return nil, nil, err
+		}
+		stmt = append(stmt, callStmt...)
+		initStmt = initStmt.Op("=")
+		if constructor.Target.PointerType != nil {
```

Comment:
Goverter should never produce code that can panic, in this case we cannot be sure if the returned pointer from the NewMethod is `nil`, therefore this use-case won't be supported.